### PR TITLE
Support ExperimentalAttribute.Message

### LINKED ISSUE #75221: Support `ExperimentlAttribute.Message`
There was a [request](https://github.com/dotnet/runtime/issues/107963) to apply experimental attribute with a message, just like `[Obsolete]`:

```C#
namespace System;

public partial class ExperimentalAttribute
{
    public string? Message { get; set; }
}
```

We don't see a problem with that but it would request feature work in the compiler to support it.

Any thoughts on this?

### PR DESCRIPTION
**Pull request by AlekseyTs**
Closes #75221

### CODE REVIEW

File: src/Compilers/CSharp/Test/Emit3/Semantics/ExperimentalAttributeTests.cs [hunk 1]
```diff
@@ -29,6 +29,8 @@ public sealed class ExperimentalAttribute : Attribute
         public ExperimentalAttribute(string diagnosticId) { }
 
         public string? UrlFormat { get; set; }
+
+        public string? Message { get; set; }
```
Comment (jcouv) [inline src/Compilers/CSharp/Test/Emit3/Semantics/ExperimentalAttributeTests.cs:33]: How should we track the update to the [spec](https://github.com/dotnet/csharplang/blob/main/proposals/csharp-12.0/experimental-attribute.md)?
One option would be to move that doc to roslyn/docs/features and update it there, as this is more a compiler feature than a language feature (we enforce regardless of language version) anyways.

Comment (AlekseyTs) [reply src/Compilers/CSharp/Test/Emit3/Semantics/ExperimentalAttributeTests.cs:33]: I can update the spec in place. I see no value in moving the document around for the API change.

**Comment by AlekseyTs** [comment]
@jcouv, @dotnet/roslyn-compiler Please review

File: src/Compilers/CSharp/Portable/Symbols/ObsoleteAttributeHelpers.cs [hunk 2]
```diff
@@ -186,12 +186,11 @@ static DiagnosticInfo createObsoleteDiagnostic(Symbol symbol, BinderFlags locati
 
                 if (data.Kind == ObsoleteAttributeKind.Experimental)
                 {
-                    Debug.Assert(data.Message is null);
                     Debug.Assert(!data.IsError);
 
                     // Provide an explicit format for fully-qualified type names.
                     return new CustomObsoleteDiagnosticInfo(MessageProvider.Instance, (int)ErrorCode.WRN_Experimental, data,
-                        new FormattedSymbol(symbol, SymbolDisplayFormat.CSharpErrorMessageFormat));
+                        new FormattedSymbol(symbol, SymbolDisplayFormat.CSharpErrorMessageFormat), string.IsNullOrEmpty(data.Message) ? "" : " - ", data.Message ?? "");
```
Comment (jcouv) [inline src/Compilers/CSharp/Portable/Symbols/ObsoleteAttributeHelpers.cs:193]: Why deviate from existing design, ie. relying on pairs of resource strings (`WRN_DeprecatedSymbolStr` corresponding to `WRN_DeprecatedSymbol` for example)? That feels safer and more flexible from a localization point of view.

Comment (AlekseyTs) [reply src/Compilers/CSharp/Portable/Symbols/ObsoleteAttributeHelpers.cs:193]: I am not following. I do not think this change violates any design and I do not think it affects flexibility or localization in any noticeable way.

Comment (AlekseyTs) [reply src/Compilers/CSharp/Portable/Symbols/ObsoleteAttributeHelpers.cs:193]: Based on offline discussion, I am going to make a change

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 2)

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler For the second review.

**Review by jjonescz — APPROVED** [review]

**Pull request merged**